homeassistant/components/fronius/__init__.py

```diff
@@ -168,12 +177,55 @@ async def _create_solar_net_device(self) -> DeviceInfo:
         )
         return solar_net_device
 
+    async def _init_devices_inverter(self) -> None:
+        """Retrieve inverter information from host."""
+        _inverter_infos: list[FroniusDeviceInfo] = []
+
+        _inverter_infos = await self._get_inverter_infos()
+
+        _LOGGER.debug("Processing inverters for: %s", _inverter_infos)
+        for _inverter_info in _inverter_infos:
+            _inverter_name = (
+                f"{DOMAIN}_inverter_{_inverter_info.solar_net_id}_{self.host}"
+            )
+
+            # Add found inverter only not already existing
+            if _inverter_info.solar_net_id in [
+                inv.inverter_info.solar_net_id for inv in self.inverter_coordinators
+            ]:
+                continue
+
+            _coordinator = FroniusInverterUpdateCoordinator(
+                hass=self.hass,
+                solar_net=self,
+                logger=_LOGGER,
+                name=_inverter_name,
+                inverter_info=_inverter_info,
+            )
+            await _coordinator.async_config_entry_first_refresh()
+            self.inverter_coordinators.append(_coordinator)
+
+            # Only for re-scans. Initial setup adds entities through sensor.async_setup_entry
+            if self.sensor_async_add_entities is not None:
+                _coordinator.add_entities_for_seen_keys(
+                    self.sensor_async_add_entities, InverterSensor
+                )
+
+            _LOGGER.info(
+                "New inverter added (UID: %s)",
+                _inverter_info.unique_id,
+            )
+
     async def _get_inverter_infos(self) -> list[FroniusDeviceInfo]:
         """Get information about the inverters in the SolarNet system."""
         try:
             _inverter_info = await self.fronius.inverter_info()
         except FroniusError as err:
-            raise ConfigEntryNotReady from err
+            if self.config_entry.state == ConfigEntryState.LOADED:
+                # During a re-scan we will attempt again as per schedule.
+                _LOGGER.warning("Re-scan failed for %s", self.host)
+            else:
+                raise ConfigEntryNotReady from err
```

Comment:
```suggestion
                raise
```
In an exception Blick you can just use `raise` to reraise the exception.

Reply:
Reverted again after testing. raise will just call the FroniusError. However, that would change the behaviour as to the setup would end up in a "setup_error" insted of a "setup_retry" state. According to the tests the desired behaviour is a re-try.

Reply:
Sorry, my bad. I've seen this wrong.